lib/overcommit/hook/pre_commit/case_conflicts.rb

```diff
@@ -0,0 +1,20 @@
+module Overcommit::Hook::PreCommit
+  # Checks for files that would conflict in case-insensitive filesystems
+  # Adapted from https://github.com/pre-commit/pre-commit-hooks
+  class CaseConflicts < Base
+    def run
+      paths = Set.new(applicable_files.map { |file| File.dirname(file) + File::SEPARATOR })
+      repo_files = Set.new(Overcommit::GitRepo.list_files(paths.to_a) + applicable_files)
```

Comment:
I think the use of `Set` for `repo_files` is still useful for the constant lookup time to see if an element is present within it. I suppose we could use the ordering of the elements returned by `git ls-tree` but that seems like more trouble than it's worth. The use of `Set#classify` works nicely here.

However, it does look like the set isn't necessary for the `paths` variable. So we can change the first line to:

```ruby
paths = applicable_files.map { |file| File.dirname(file) + File::SEPARATOR }
```

...allowing us to get rid of the `.to_a`.